lib/puppet/type/wildfly_deployment.rb

```diff
@@ -54,6 +55,11 @@
     end
   end
 
+  newparam(:undeploy_first, :boolean => true, :parent => Puppet::Parameter::Boolean) do
```

Comment:
Naming things is hard!  `undeploy_first` was the best I came up with.  Open to suggestions. :)